platforms/core-configuration/base-diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DiagnosticsTaskNames.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.plugins.HelpTasksPlugin;
+
+/**
+ * The names of the diagnostic tasks added by the {@link HelpTasksPlugin} and the
+ * {@code SoftwareReportingTasksPlugin}.
+ *
+ * @since 8.13
+ */
+@Incubating
+public interface DiagnosticsTaskNames {
+    /**
+     * The name of the help task group.
+     *
+     * @since 8.13
+     */
+    String HELP_GROUP = "help";
+
+    /**
+     * The name of the properties report task.
+     *
+     * @since 8.13
+     */
+    String PROPERTIES_TASK = "properties";
+
+    /**
+     * The name of the dependencies report task.
+     *
+     * @since 8.13
+     */
+    String DEPENDENCIES_TASK = "dependencies";
+
+    /**
+     * The name of the dependency insight report task.
+     *
+     * @since 8.13
+     */
+    String DEPENDENCY_INSIGHT_TASK = "dependencyInsight";
```

Comment:
Seems odd that the names of dependency management/software report tasks live in the `DiagnosticsTaskNames` for the base diagnostics project

Reply:
Yes it is, but I want to have a single source of truth for them, and the `HelpTasksPlugin` needs to know about them to avoid a change to the public API.

As a followup, I think we should deprecate accessing these through the `HelpTasksPlugin` and eventually move the definitions to `SoftwareReportingTasksPlugin` directly.

It has to be public API itself, to avoid the "public API shouldn't extend internal types" warning.  But it can be deprecated too.  I can do that immediately after merging this change, I just don't want to also add deprecation warnings in this PR.